packages/resolvers/default/src/DefaultResolver.js

```diff
@@ -179,6 +179,33 @@ class NodeResolver {
       return builtin;
     }
 
+    // Resolve peer dependencies
+    // Finds the first package that does not specify filename as a peer dependency
+    // and resolve from there. If there was no package, or all packages specified
+    // filename as a peer dependency, the module is marked as external
+    let pkg = await this.findPackage(dir);
```

Comment:
We should probably have a cache in front of `findPackage`?

Reply:
There is a cache in findPackage > readPackage, or do you want to prevent traveling up the directory hierarchy?